glide-core/telemetry/src/metrics_exporter_file.rs

```diff
@@ -0,0 +1,243 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use opentelemetry_sdk::metrics::data::ResourceMetrics;
+use opentelemetry_sdk::metrics::data::{Gauge, Histogram, Sum};
+use opentelemetry_sdk::metrics::exporter::PushMetricExporter;
+use opentelemetry_sdk::metrics::MetricError;
+use opentelemetry_sdk::metrics::MetricResult;
+use opentelemetry_sdk::metrics::Temporality;
+use serde_json::{Map, Value};
+use std::any::Any;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::PathBuf;
+use std::result::Result;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// An OpenTelemetry exporter that writes Metrics to a file on export.
+pub struct FileMetricExporter {
+    is_shutdown: AtomicBool,
+    path: PathBuf,
+}
+
+impl FileMetricExporter {
+    /// Creates a new FileMetricExporter that writes metrics to the specified path.
+    ///
+    /// # Arguments
+    /// * `path` - The path where metrics will be written. This can be either a file or directory path.
+    ///
+    /// # Behavior
+    /// - If the path points to a directory:
+    ///   - The directory must exist
+    ///   - Metrics will be written to a file named "signals.json" within that directory
+    /// - If the path points to a file:
+    ///   - If the file exists, new metrics will be appended to it (existing data is preserved)
+    ///   - If the file doesn't exist, it will be created
+    ///   - The parent directory must exist
+    ///
+    /// # Errors
+    /// Returns a MetricError if:
+    /// - The parent directory doesn't exist
+    /// - The path points to a directory that doesn't exist
+    /// - The user doesn't have write permissions for the target location
+    pub fn new(path: PathBuf) -> Result<Self, MetricError> {
+        Ok(Self {
+            is_shutdown: AtomicBool::new(false),
+            path,
+        })
+    }
+}
+
+#[async_trait]
+impl PushMetricExporter for FileMetricExporter {
+    fn temporality(&self) -> Temporality {
+        Temporality::Cumulative
+    }
+
+    async fn export(&self, metrics: &mut ResourceMetrics) -> MetricResult<()> {
+        if self.is_shutdown.load(Ordering::SeqCst) {
+            return Err(MetricError::Other("Exporter is shutdown".to_string()));
+        }
+
+        let mut data_file = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&self.path)
+            .map_err(|err| MetricError::Other(format!("Unable to open exporter file: {err}")))?;
+
+        let metrics_json = to_json(metrics)
+            .map_err(|e| MetricError::Other(format!("Failed to serialize metrics to JSON: {e}")))?;
+        let json_string = serde_json::to_string(&metrics_json)
+            .map_err(|e| MetricError::Other(format!("Failed to serialize metrics to JSON: {e}")))?;
+
+        data_file
+            .write(format!("{}\n", json_string).as_bytes())
+            .map_err(|e| MetricError::Other(format!("File write error: {e}")))?;
```

Comment:
```suggestion
        writeln!(data_file, "{}", json_string)
            .map_err(|e| MetricError::Other(format!("File write error: {e}")))?;
```

(add `use std::io::Write;`)